src/qs8-packw/gen/qs8-packw-x8c4-gemm-goi-avxvnni-prfm.c

```diff
@@ -0,0 +1,412 @@
+        const __m256i v46_02 = _mm256_unpacklo_epi64(v45_0145, v67_0145);
+        const __m256i v46_13 = _mm256_unpackhi_epi64(v45_0145, v67_0145);
+        const __m256i v57_02 = _mm256_unpacklo_epi64(v45_2367, v67_2367);
+        const __m256i v57_13 = _mm256_unpackhi_epi64(v45_2367, v67_2367);
+
+       const __m256i v04_0 = _mm256_permute2f128_si256(v02_02, v46_02, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v04_1 = _mm256_permute2f128_si256(v02_02, v46_02, _MM_SHUFFLE(0, 3, 0, 1));
+       const __m256i v15_0 = _mm256_permute2f128_si256(v02_13, v46_13, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v15_1 = _mm256_permute2f128_si256(v02_13, v46_13, _MM_SHUFFLE(0, 3, 0, 1));
+       const __m256i v26_0 = _mm256_permute2f128_si256(v13_02, v57_02, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v26_1 = _mm256_permute2f128_si256(v13_02, v57_02, _MM_SHUFFLE(0, 3, 0, 1));
+       const __m256i v37_0 = _mm256_permute2f128_si256(v13_13, v57_13, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v37_1 = _mm256_permute2f128_si256(v13_13, v57_13, _MM_SHUFFLE(0, 3, 0, 1));
+
+        xnn_prefetch_to_l1((const int8_t*) w0 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w1 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w2 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w3 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w4 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w5 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w6 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w7 + 448);
+
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v04_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v15_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v26_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v37_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v04_1);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v15_1);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v26_1);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v37_1);
+
+        _mm256_storeu_si256((__m256i *)&out[0],  v04_0);
+        _mm256_storeu_si256((__m256i *)&out[32],  v15_0);
+        _mm256_storeu_si256((__m256i *)&out[64],  v26_0);
+        _mm256_storeu_si256((__m256i *)&out[96],  v37_0);
+        _mm256_storeu_si256((__m256i *)&out[128],  v04_1);
+        _mm256_storeu_si256((__m256i *)&out[160],  v15_1);
+        _mm256_storeu_si256((__m256i *)&out[192],  v26_1);
+        _mm256_storeu_si256((__m256i *)&out[224],  v37_1);
+
+        w0 += 32;
+        w1 += 32;
+        w2 += 32;
+        w3 += 32;
+        w4 += 32;
+        w5 += 32;
+        w6 += 32;
+        w7 += 32;
+        out += 256;
+      }
+
+      // KC main loop multiple of 8x4
+      for (; k >= 4; k -= 4) {
+        __m256i v0 = _mm256_set1_epi32((int32_t) unaligned_load_u32(w0));
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w1)), 0x02);
```

Comment:
on a future avx10 version of this kernel, this and the remainder code could use a kmask to read and embedded broadcast to the correct channels.